test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/libPopFrameTest.cpp

```diff
@@ -152,20 +154,25 @@ Java_PopFrameTest_popFrame(JNIEnv *jni, jclass cls, jthread thread) {
 
 JNIEXPORT void JNICALL
 Java_PopFrameTest_ensureAtBreakpoint(JNIEnv *jni, jclass cls) {
-  bool need_stop = false;
-
   LOG("Main: ensureAtBreakpoint\n");
-  while (!need_stop) {
-    RawMonitorLocker rml(jvmti, jni, monitor);
-    need_stop = bp_sync_reached;
-    sleep_ms(1); // 1 millisecond
+  RawMonitorLocker rml(jvmti, jni, monitor);
+  int attempts = 0;
+  while (!bp_sync_reached) {
+    LOG("Main: ensureAtBreakpoint: waitig 5 millis\n");
+    if (++attempts > 100) {
+      fatal(jni, "Main: ensureAtBreakpoint: waited 1 sec");
```

Comment:
1 second isn't very long when you are talking about something that is relying on debugger/debuggee communication. Maybe wait 100ms at a time for a total of 10 seconds.

Reply:
Thanks for the comments, Chris and Dan. Updated as Chris suggested. I've added this with `-Xcomp` consideration as the worst case scenario in mind. Now, I think it is more save to make it 10 seconds instead of one. Is it going to be good enough? In fact, I've added this for manual testing to save time in waiting for test completion when it is deadlocked.

Reply:
runtime/8176717/TestInheritFD.java has an example of what I'm talking about:
```
    public static float timeoutFactor = Float.parseFloat(System.getProperty("test.timeout.factor", "1.0"));
    public static long subProcessTimeout = (long)(15L * timeoutFactor);
```
so you fetch the test.timeout.factor value and then you scale your delay value.

Also:

nit typo: s/waitig/waiting/

Reply:
Thank you for the example and for catching the typo. The timeout factor also needs to be passed to the native side. I think, this fragment is not worth this kind of extra complexity. One approach would be to just make it big enough, eg. make it 100 seconds instead of 10.  Another - to get rid of this trap at all. What would you prefer?

Reply:
So that would mean that the native side would always wait for 100 seconds?
Or will it wait for some increment of time upto a maximum of 100 seconds?

Reply:
It wait for some increment of time upto a maximum of 100 seconds.

Reply:
I'm good with that. Thanks for clarifying.

Reply:
Thanks, Dan.
I've decided to increase this timeout to 20 secs. Pushed now.